cleanlab_studio/utils/data_enrichment/enrich.py

```diff
@@ -1,91 +1,71 @@
-from typing import Any, List, Optional, Tuple, Union
-import pandas as pd
+from typing import Any, List, Tuple, Union, Dict
+from functools import lru_cache
 from cleanlab_studio.internal.enrichment_utils import (
-    extract_df_subset,
     get_prompt_outputs,
     get_regex_match_or_replacement,
     get_constrain_outputs_match,
     get_optimized_prompt,
-    Replacement,
 )
+from cleanlab_studio.studio.enrichment import EnrichmentOptions
 
-from cleanlab_studio.studio.studio import Studio
-
-
-def enrich_data(
-    studio: Studio,
-    data: pd.DataFrame,
-    prompt: str,
-    *,
-    regex: Optional[Union[str, Replacement, List[Replacement]]] = None,
-    constrain_outputs: Optional[List[str]] = None,
-    optimize_prompt: bool = True,
-    subset_indices: Optional[Union[Tuple[int, int], List[int]]] = (0, 3),
-    new_column_name: str = "metadata",
-    disable_warnings: bool = False,
-    **kwargs: Any,
-) -> pd.DataFrame:
+
+@lru_cache(maxsize=None)
+def _get_pandas():
+    import pandas as pd
+
+    return pd
+
+
+@lru_cache(maxsize=None)
+def _get_tqdm():
+    from tqdm import tqdm
+
+    return tqdm
+
```

Comment:
what is going on here?

tqdm is already a dependency of this package, there should be no special logic to lazy import it

https://github.com/cleanlab/cleanlab-studio/blob/c2a3013fbb1b2a77d3182a7894367b357a5b9c35/setup.py#L57